### PR TITLE
Manual: Updates for 0.4.0

### PR DESCRIPTION
**Pull request by ax3l**
- add `tbg` section
- update "basic usage" section:
  - fix outdated .cfg file names
  - improve clarity and structure
- update install instructions:
  - refresh version ranges
  - update min. alpaka
  - add notes, e.g. to spack and use nvidia docker 2+
  - replace wget with more widespread curl command (linux/osx)
  - remove untar verbosity

Greetings from Toronto Airport!

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -89,13 +89,16 @@ zlib
 
 boost
 """""
-- 1.62.0-1.64.0 (``program_options``, ``regex`` , ``filesystem``, ``system``, ``math``, ``serialization`` and header-only libs, optional: ``fiber`` with ``context``, ``thread``, ``chrono``, ``atomic``, ``date_time``)
-- download from `http://www.boost.org <http://sourceforge.net/projects/boost/files/boost/1.62.0/boost_1_62_0.tar.gz/download>`_
+- 1.62.0 - 1.68.0 (``program_options``, ``regex`` , ``filesystem``, ``system``, ``math``, ``serialization`` and header-only libs, optional: ``fiber`` with ``context``, ``thread``, ``chrono``, ``atomic``, ``date_time``)
```
Comment (ax3l) [inline INSTALL.rst]: Help wanted: we should verify that boost 1.67.0 and 1.68.0 work for us, especially with nvcc

File: docs/source/usage/tbg.rst [hunk 2]
```diff
@@ -3,15 +3,29 @@
 TBG
 ===
 
-.. sectionauthor:: Axel Huebl, Richard Pausch
```
Comment (ax3l) [inline docs/source/usage/tbg.rst]: Moved below to shared section

**Comment by ax3l** [comment]
Cross-link: Alpaka Boost 1.68.0 support status https://github.com/ComputationalRadiationPhysics/alpaka/pull/606

File: docs/source/usage/tbg.rst [hunk 3]
```diff
@@ -3,15 +3,29 @@
...
-todo: explain idea and use case
+Our tool *template batch generator* (``tbg``) abstracts program runtime options from technical details of supercomputers.
+On a desktop PC, one can just execute a command interactively and instantaneously.
+Contrarily on a supercomputer, resources need to be shared between different users efficiently via *job scheduling*.
+Scheduling on today's supercomputers is usually done via *batch systems* that define various queues of resources.
+
+An unfortunate aspect about batch systems from a user's perspective is, that their usage varies a lot.
+And naturally, different systems have different resources in queues that need to be described.
+
+We abstract the description of queues, resource acquisition and job submission away from PIConGPU user input via *template files* (``.tpl``).
```
Comment (sbastrakov) [inline docs/source/usage/tbg.rst]: I feel it's a little imprecise: `queues, resource acquisition and job submission` are not part of `PIConGPU user input` per se, but part of a launch command line. Maybe just remove words `from PIConGPU user input` ?

File: docs/source/usage/tbg.rst [hunk 4]
```diff
@@ -3,15 +3,29 @@
...
+For example, the ``.cfg`` file defines how many *devices* shall be used for computation, but the ``.tpl`` file calculates how many *physical nodes* will be requested.
```
Comment (sbastrakov) [inline docs/source/usage/tbg.rst]: I think here `the ` .cfg file is rather unnatural, we are talking in general about a configuration file, right?

File: INSTALL.rst (hunk 1 above)
Comment (ax3l) [reply INSTALL.rst]: 1.67.0 seems to be working, 1.68.0 is problematic (see alpaka link below)

File: docs/source/usage/tbg.rst [hunk 5]
```diff
@@ -14,8 +14,9 @@ Scheduling on today's supercomputers is usually done via *batch systems* that de
 An unfortunate aspect about batch systems from a user's perspective is, that their usage varies a lot.
 And naturally, different systems have different resources in queues that need to be described.
 
-We abstract the description of queues, resource acquisition and job submission away from PIConGPU user input via *template files* (``.tpl``).
-For example, the ``.cfg`` file defines how many *devices* shall be used for computation, but the ``.tpl`` file calculates how many *physical nodes* will be requested.
+PIConGPU runtime options are described in *configuration files* (``.cfg``).
+We abstract the description of queues, resource acquisition and job submission via *template files* (``.tpl``).
+For example, a ``.cfg`` file defines how many *devices* shall be used for computation, but a ``.tpl`` file calculates how many *physical nodes* will be requested.
 Also, the ``.tpl`` file takes care of how to spawn a process when scheduled, e.g. with ``mpiexec`` and which flags for networking details need to be passed.
```
Comment (sbastrakov) [inline docs/source/usage/tbg.rst]: Sorry, did not mark it last time. Here it also has to be `a` tpl file.

**Comment by ax3l** [comment]
@PrometheusPi ping :)

**Pull request merged**